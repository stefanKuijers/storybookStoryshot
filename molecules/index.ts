export { default as Avatar } from "./Avatar/Avatar";

export { default as Dropdown } from "./Dropdown/Dropdown";
export { default as DropdownItem } from "./DropdownItem/DropdownItem";
export { default as DropdownItemNavLink } from "./DropdownItemNavLink/DropdownItemNavLink";
