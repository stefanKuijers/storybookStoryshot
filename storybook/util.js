const appRoot = require("app-root-path");
const path = require("path");

// const withTests = require('@storybook/addon-jest').withTests;
// // import { withTests } from ;
// const testResults = require('../test/unit/.jest-test-results.json');

function parseInt(str, fallback) {
  return (str && Number.parseInt(str, 10)) || fallback;
}

function resolveAbsolute(...paths) {
  return path.resolve(appRoot.path, path.join(...paths));
}

// function stageStoriesOf(name, components = []) {
// 	return storiesOf(name, module)
//     .addDecorator(withKnobs)
//     .addDecorator(stageWithTests(components))
// 	  .addDecorator(host({
// 	    title: 'A host container for components under test.',
// 	    align: 'center bottom',
// 	    height: '80%',
// 	    width: '80%',
// 	    padding: 15
// 	  }));
// }

// function stageWithTests(components) {
//   return withTests({
//     testResults
//   })(components);
// }

module.exports = {
  parseInt,
  resolveAbsolute
  // stageStoriesOf,
  // stageWithTests
};
