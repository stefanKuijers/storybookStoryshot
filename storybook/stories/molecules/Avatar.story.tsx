import { text } from "@storybook/addon-knobs";
import React from "react";

import { Avatar } from "../../../molecules";
import { stageStoriesOf } from "../utils";

stageStoriesOf("Molecules/Avatar", "Avatar")
  .add("Simple Avatar", () => {
    return (
      <>
        <Avatar label={text("label", "DL")} />
        <Avatar name={text("User's name", "John Doe")} />
      </>
    );
  })
  .add("Avatar with info", () => {
    return (
      <>
        <Avatar
          name={text("User' name", "John Doe")}
          userName={text("Username", "john88")}
          emailAddress={text("Email address", "john88@mail.com")}
        />
      </>
    );
  });
