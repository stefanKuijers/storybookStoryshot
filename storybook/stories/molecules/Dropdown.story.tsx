import { boolean, text } from "@storybook/addon-knobs";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Placement } from "reactstrap/lib/Popper";

import { Title } from "../../../atoms";
import { Dropdown, DropdownItem, DropdownItemNavLink } from "../../../molecules";
import { stageStoriesOf, t } from "../utils";

const dropdownItems = [
  {
    children: "Item 1",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 2",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 3",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 4",
    type: DropdownItem,
    options: {}
  }
];

const headerMenuDropdownItems = [
  {
    type: DropdownItemNavLink,
    children: "Active Projects (13)",
    options: {
      to: "/projects/active",
      icon: "active"
    }
  },
  {
    type: DropdownItemNavLink,
    children: "Removed Projects (1)",
    options: {
      to: "/projects/removed",
      icon: "trash"
    }
  },
  {
    type: DropdownItemNavLink,
    children: "Archived Projects (432)",
    options: {
      to: "/projects/archived",
      icon: "archived"
    }
  }
];

const placement: Placement = "auto";

stageStoriesOf("Molecules/Dropdown", "Dropdown")
  .add("Dropdown", () => (
    <Dropdown
      id="DropdownDefaultId"
      content={text("Custom child", "Lorem ipsum")}
      dropdownItems={dropdownItems}
      placement={placement}
      containerClassName={text("Container class name?", "")}
      className={text("Custom class?", "")}
      showArrow={boolean("Show arrow?", false)}
    />
  ))
  .add("Header Dropdown", () => (
    <Router>
      <Dropdown
        id={"headerDropdown"}
        content={<Title className={"HeaderTitle"} type={"h1"} title={t("This stuff")} />}
        dropdownItems={headerMenuDropdownItems}
        showArrow
      />
    </Router>
  ));
