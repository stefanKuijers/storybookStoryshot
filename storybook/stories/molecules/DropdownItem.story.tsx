import { boolean, text } from "@storybook/addon-knobs";
import React from "react";

import { DropdownItem } from "../../../molecules";
import { stageStoriesOf } from "../utils";

stageStoriesOf("Molecules/Dropdown", "DropdownItem").add("DropdownItem", () => (
  <DropdownItem isSeparator={boolean("Is separator?", false)} icon={"fire"} onClick={console.log}>
    {text("Custom child?", "Lorem ipsum")}
  </DropdownItem>
));
