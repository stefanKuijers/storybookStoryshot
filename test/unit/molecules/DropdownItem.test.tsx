import React from "react";
import renderer from "react-test-renderer";

import { DropdownItem } from "../../../molecules";
import { t } from "../mocks/util";

describe("DropdownItem", () => {
  it("matches the snapshot", () => {
    const Renderer = renderer.create(
      <DropdownItem isSeparator={false} icon={"fire"}>
        {t("Lorem ipsum")}
      </DropdownItem>
    );

    expect(Renderer.toJSON()).toMatchSnapshot();
  });
});
