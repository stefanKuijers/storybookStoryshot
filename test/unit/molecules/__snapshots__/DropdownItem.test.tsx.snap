// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem matches the snapshot 1`] = `
<div
  className="DropdownItem tclass-dropdownItem"
>
  <a>
    <figure
      className="Icon tclass-icon Icon__fire DropdownItem__icon"
    >
      <span
        className="Icon__holder"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test-file-stub",
          }
        }
      />
    </figure>
    <span
      className="DropdownItem__content"
    >
      Lorem ipsum
    </span>
  </a>
</div>
`;
